felix/calc/policy_resolver.go

```diff
@@ -129,6 +140,12 @@ func (pr *PolicyResolver) markEndpointsMatchingPolicyDirty(polKey model.PolicyKe
 
 func (pr *PolicyResolver) OnPolicyMatch(policyKey model.PolicyKey, endpointKey interface{}) {
 	log.Debugf("Storing policy match %v -> %v", policyKey, endpointKey)
+	// If it's first time the policy become matched, add it to the tier
+	if _, ok := pr.policySorter.tier.Policies[policyKey]; !ok {
```

Comment:
Suggest adding a method to `policySorter` to do this check: `if !policySorter.HasPolicy(policyKey) {` for example